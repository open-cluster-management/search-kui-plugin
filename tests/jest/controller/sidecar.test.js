/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
// Copyright (c) 2021 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'
import "regenerator-runtime"
const fn = require('../../../dist/controller/sidecar')
const related = require('../../data/related')

const data = related.data
const resource = related.yaml

describe('Building Sidecar', () => {
  it('should build a resource sidecar with tabs [Summary, YAML, Related Resources]', () => {
    expect(fn.buildSidecar('resource', data, resource)).toMatchSnapshot()
  })

  it('should build a query sidecar with tabs [Related Resources]', () => {
    expect(fn.buildSidecar('query', data)).toMatchSnapshot()
  })
})
