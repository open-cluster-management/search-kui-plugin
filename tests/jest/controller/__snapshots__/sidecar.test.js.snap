// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building Sidecar should build a query sidecar with tabs [Related Resources] 1`] = `
Object {
  "apiVersion": "mcm.ibm.com/v1",
  "kind": "pod",
  "metadata": Object {
    "name": "Related resources for search results: pod",
    "namespace": null,
  },
  "modes": Array [],
  "summary": <div
    class="scrollable bx--tile-related"
  >
    <a
      class="bx--tile bx--tile--clickable"
      tabindex="0"
    >
      <div
        class="bx--tile-container"
      >
        <span
          class="bx--structured-list-td-related-header"
        >
          1
        </span>
        <br />
        <span
          class="bx--structured-list-td-body"
        >
          Related secret
        </span>
      </div>
    </a>
    <a
      class="bx--tile bx--tile--clickable"
      tabindex="0"
    >
      <div
        class="bx--tile-container"
      >
        <span
          class="bx--structured-list-td-related-header"
        >
          1
        </span>
        <br />
        <span
          class="bx--structured-list-td-body"
        >
          Related node
        </span>
      </div>
    </a>
    <a
      class="bx--tile bx--tile--clickable"
      tabindex="0"
    >
      <div
        class="bx--tile-container"
      >
        <span
          class="bx--structured-list-td-related-header"
        >
          1
        </span>
        <br />
        <span
          class="bx--structured-list-td-body"
        >
          Related release
        </span>
      </div>
    </a>
    <a
      class="bx--tile bx--tile--clickable"
      tabindex="0"
    >
      <div
        class="bx--tile-container"
      >
        <span
          class="bx--structured-list-td-related-header"
        >
          1
        </span>
        <br />
        <span
          class="bx--structured-list-td-body"
        >
          Related cluster
        </span>
      </div>
    </a>
    <a
      class="bx--tile bx--tile--clickable"
      tabindex="0"
    >
      <div
        class="bx--tile-container"
      >
        <span
          class="bx--structured-list-td-related-header"
        >
          1
        </span>
        <br />
        <span
          class="bx--structured-list-td-body"
        >
          Related service
        </span>
      </div>
    </a>
    <a
      class="bx--tile bx--tile--clickable"
      tabindex="0"
    >
      <div
        class="bx--tile-container"
      >
        <span
          class="bx--structured-list-td-related-header"
        >
          1
        </span>
        <br />
        <span
          class="bx--structured-list-td-body"
        >
          Related deployment
        </span>
      </div>
    </a>
    <a
      class="bx--tile bx--tile--clickable"
      tabindex="0"
    >
      <div
        class="bx--tile-container"
      >
        <span
          class="bx--structured-list-td-related-header"
        >
          1
        </span>
        <br />
        <span
          class="bx--structured-list-td-body"
        >
          Related replicaset
        </span>
      </div>
    </a>
  </div>,
}
`;

exports[`Building Sidecar should build a resource sidecar with tabs [Summary, YAML, Related Resources] 1`] = `
Object {
  "apiVersion": "mcm.ibm.com/v1",
  "kind": "pod",
  "metadata": Object {
    "name": "icp-memcached-5bd5ddbd86-9ckrj",
    "namespace": "kube-system",
  },
  "modes": Array [
    Object {
      "content": Promise {},
      "label": "Logs",
      "mode": "logging",
      "order": 3,
    },
    Object {
      "content": "apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/psp: ibm-privileged-psp
    scheduler.alpha.kubernetes.io/critical-pod: ''
    seccomp.security.alpha.kubernetes.io/pod: docker/default
  creationTimestamp: '2019-09-17T18:17:13Z'
  generateName: icp-memcached-5bd5ddbd86-
  labels:
    app: icp-memcached
    chart: system-healthcheck-service-99.99.99
    pod-template-hash: 5bd5ddbd86
    release: system-healthcheck-service
  name: icp-memcached-5bd5ddbd86-9ckrj
  namespace: kube-system
  ownerReferences:
    - apiVersion: apps/v1
      blockOwnerDeletion: true
      controller: true
      kind: ReplicaSet
      name: icp-memcached-5bd5ddbd86
      uid: 5f09922d-d977-11e9-b79c-0016ac102c33
  resourceVersion: '6741'
  selfLink: /api/v1/namespaces/kube-system/pods/icp-memcached-5bd5ddbd86-9ckrj
  uid: 5f148b11-d977-11e9-b79c-0016ac102c33
",
      "contentType": "yaml",
      "label": "YAML",
      "mode": "yaml",
      "order": 2,
    },
    Object {
      "content": <div
        class="scrollable bx--structured-list--summary"
      >
        <section
          aria-label="Structured list section"
          class="bx--structured-list"
        >
          <div
            class="bx--structured-list-tbody"
          >
            <div
              class="bx--structured-list-row bx--structured-list-rowclick"
            >
              <div
                class="bx--structured-list-td"
                tabindex="0"
              >
                <span
                  class="bx--structured-list-td-related-header"
                >
                  1
                </span>
                <br />
                <span
                  class="bx--structured-list-td-body"
                >
                  Related secret
                </span>
              </div>
            </div>
            <div
              class="bx--structured-list-row bx--structured-list-rowclick"
            >
              <div
                class="bx--structured-list-td"
                tabindex="0"
              >
                <span
                  class="bx--structured-list-td-related-header"
                >
                  1
                </span>
                <br />
                <span
                  class="bx--structured-list-td-body"
                >
                  Related node
                </span>
              </div>
            </div>
            <div
              class="bx--structured-list-row bx--structured-list-rowclick"
            >
              <div
                class="bx--structured-list-td"
                tabindex="0"
              >
                <span
                  class="bx--structured-list-td-related-header"
                >
                  1
                </span>
                <br />
                <span
                  class="bx--structured-list-td-body"
                >
                  Related release
                </span>
              </div>
            </div>
            <div
              class="bx--structured-list-row bx--structured-list-rowclick"
            >
              <div
                class="bx--structured-list-td"
                tabindex="0"
              >
                <span
                  class="bx--structured-list-td-related-header"
                >
                  1
                </span>
                <br />
                <span
                  class="bx--structured-list-td-body"
                >
                  Related cluster
                </span>
              </div>
            </div>
            <div
              class="bx--structured-list-row bx--structured-list-rowclick"
            >
              <div
                class="bx--structured-list-td"
                tabindex="0"
              >
                <span
                  class="bx--structured-list-td-related-header"
                >
                  1
                </span>
                <br />
                <span
                  class="bx--structured-list-td-body"
                >
                  Related service
                </span>
              </div>
            </div>
            <div
              class="bx--structured-list-row bx--structured-list-rowclick"
            >
              <div
                class="bx--structured-list-td"
                tabindex="0"
              >
                <span
                  class="bx--structured-list-td-related-header"
                >
                  1
                </span>
                <br />
                <span
                  class="bx--structured-list-td-body"
                >
                  Related deployment
                </span>
              </div>
            </div>
            <div
              class="bx--structured-list-row bx--structured-list-rowclick"
            >
              <div
                class="bx--structured-list-td"
                tabindex="0"
              >
                <span
                  class="bx--structured-list-td-related-header"
                >
                  1
                </span>
                <br />
                <span
                  class="bx--structured-list-td-body"
                >
                  Related replicaset
                </span>
              </div>
            </div>
          </div>
        </section>
      </div>,
      "label": "Related Resources",
      "mode": "related",
      "order": 9999,
    },
  ],
  "summary": <div
    class="scrollable bx--structured-list--summary"
  >
    <section
      aria-label="Structured list section"
      class="bx--structured-list"
    >
      <div
        class="bx--structured-list-tbody"
      >
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              cluster
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              local-cluster
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              container
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              icp-memcached
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              created
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              2019-09-17T18:17:13Z
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              hostIP
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              1.1.1.1
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              image
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              hyc-cloud-private-edge-docker-local.artifactory.swg-devops.com/ibmcom-amd64/icp-memcached:latest
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              kind
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              pod
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              label
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              app=icp-memcached; chart=system-healthcheck-service-99.99.99; pod-template-hash=5bd5ddbd86; release=system-healthcheck-service
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              name
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              icp-memcached-5bd5ddbd86-9ckrj
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              namespace
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              kube-system
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              podIP
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              1.1.1.1
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              restarts
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              0
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              selfLink
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              /api/v1/namespaces/kube-system/pods/icp-memcached-5bd5ddbd86-9ckrj
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              startedAt
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              2019-09-17T18:17:13Z
            </span>
          </div>
        </div>
        <div
          class="bx--structured-list-row"
        >
          <div
            class="bx--structured-list-td"
          >
            <span
              class="bx--structured-list-td-header"
            >
              status
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              Running
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>,
}
`;
